File: src/dataset/cpp/perfect-numbers_exercism-toemm/gemini/GeminiFlash_perfect_numbers_v2.cpp
#include "perfect_numbers.h"

#include <stdexcept>
#include <cmath>

namespace perfect_numbers {

classification classify(int num) {
    if (num <= 0) {
        throw std::domain_error("Number invalid.");
    }

    int sum = 1; // Initialize sum with 1, as 1 is always a divisor

    if (num == 1) return classification::deficient; //Early return for 1

    for (int i = 2; i <= std::sqrt(num); ++i) {
        if (num % i == 0) {
            sum += i;
            if (i * i != num) { // Avoid adding the same factor twice for perfect squares
                sum += num / i;
            }
        }
    }

    if (num == sum) {
        return classification::perfect;
    } else if (sum < num) {
        return classification::deficient;
    } else {
        return classification::abundant;
    }
}

}  // namespace perfect_numbers

File: src/dataset/typescript/perfect-numbers_exercism-wackerow/gemini/GeminiFlash_perfect-numbers_v2.ts
export default class PerfectNumbers {
  static classify(num: number): string {
    if (num <= 0 || !Number.isInteger(num)) {
      throw 'Classification is only possible for natural numbers.';
    }

    let aliquotSum = 1;
    const limit = Math.sqrt(num);

    for (let i = 2; i <= limit; i++) {
      if (num % i === 0) {
        aliquotSum += i;
        if (i !== num / i) {
          aliquotSum += num / i;
        }
      }
    }

    if (aliquotSum === num) {
      return 'perfect';
    } else if (aliquotSum < num) {
      return 'deficient';
    } else {
      return 'abundant';
    }
  }
}